lib/mqtt_abstraction_impl.cpp

```diff
@@ -296,18 +295,10 @@ void MQTTAbstractionImpl::on_mqtt_message(std::shared_ptr<Message> message) {
         std::unique_lock<std::mutex> lock(handlers_mutex);
         std::vector<Handler> local_handlers;
         for (auto& [handler_topic, handler] : this->message_handlers) {
-            bool topic_matches = false;
-            if (is_everest_topic) {
-                // everest topics never contain wildcards, so a direct comparison is enough
-                if (topic == handler_topic) {
-                    topic_matches = true;
-                }
-            } else {
-                topic_matches = MQTTAbstractionImpl::check_topic_matches(topic, handler_topic);
-            }
```

Comment:
This is more expensive.  Are the wildcards necessary within the everest prefix namespace?

Reply:
I don't really see a usecase where you want to subscribe (or publish) to any topics within the everest prefix namespace (with or without wildcards) from an external mqtt client. We [typically discourage this](https://lfenergy.zulipchat.com/#narrow/channel/417677-EVerest.3A-Car-com.2E/topic/.E2.9C.94.20failed.20on.20cable.20check/near/445412041) as well, especially since the internal communication has changed in the past and will in the future

Reply:
One of the intended use cases is to support wildcard on everest topics.
(e.g. to listen to session events from any EVSE manager).
The existing design doesn't make it easy to set an optional flag to cover this case other than perhaps using a global variable which I don't like as a solution.

Reply:
currently there is a check for the topic matching mqtt_everest_prefix
```
        if (topic.find(mqtt_everest_prefix) == 0) {
```

i.e. if the topic starts `everest` which would also include `everest_api` then wildcards are disabled.

There is existing use of the external mqtt API to publish to `everest_api` in the Setup module.
Currently `everest_api` is prevented from supporting wildcard matches. 

If the aim is to not support wildcard under `everest/` then the check can be improved however that would prevent being able to subscribe to all session events from all EVSE managers.

Reply:
We should probably discuss this _offline_

Reply:
> currently there is a check for the topic matching mqtt_everest_prefix
> 
> ```
>         if (topic.find(mqtt_everest_prefix) == 0) {
> ```
> 
> i.e. if the topic starts `everest` which would also include `everest_api` then wildcards are disabled.
> 
> There is existing use of the external mqtt API to publish to `everest_api` in the Setup module. Currently `everest_api` (`everest_external` ...) are prevented from supporting wildcard matches.

This should not be a problem at the moment because the check is for `everest/` (the forward slash at the end is always added to the prefix here: https://github.com/EVerest/everest-framework/blob/bb3d3a91bb50031d21aa3d43220801a9eb69a6bd/lib/runtime.cpp#L334 ), so `everest_api` etc. would be able to support wildcard matching without this change

Reply:
ah - not spotted the addition of `/` it looked like it was set from a YAML config file.
Anyway as for this PR - I'll put that check back - but it would be useful to have the ability to subscribe for all session events. It would be all too easy for a missing connection in the YAML config file and events would not be seen.

Since 128 managers can be configured that is 128 separate subscribes that could instead be handled by a single wildcard.
```
  evse_manager:
    interface: evse_manager
    min_connections: 1
    max_connections: 128
```

Reply:
We could think about adding some form of wildcard / "get all requirements of a certain interface" functionality directly to the framework (that you could enable in the manifest/config kind of like the global errors). But I would try to avoid poking around in the mqtt communication between modules

Reply:
I've raised the following improvement suggestion to capture this: #217